packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -348,4 +376,35 @@ export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchain
     async getNFTs() {
         return this.crossmintService.fetchNFTs(this.account.address, this.chain);
     }
+
+    hasEIP1559Support() {
+        const chainsNotSupportingEIP1559: EVMBlockchainIncludingTestnet[] = [
+            EVMBlockchainIncludingTestnet.ZKYOTO,
+            EVMBlockchainIncludingTestnet.ASTAR_ZKEVM,
+        ];
+        return !chainsNotSupportingEIP1559.includes(this.chain);
+    }
+
+    getLegacyTransactionFeesParamsIfApply(gasFeeParams?: GasFeeTransactionParams) {
+        const { maxFeePerGas, maxPriorityFeePerGas } = gasFeeParams ?? {};
+
+        if (this.hasEIP1559Support()) {
+            return {
+                maxFeePerGas: maxFeePerGas ? BigInt(maxFeePerGas.toString()) : undefined,
+                maxPriorityFeePerGas: maxPriorityFeePerGas ? BigInt(maxPriorityFeePerGas.toString()) : undefined,
+            };
+        } else {
+            if (maxFeePerGas || maxPriorityFeePerGas) {
+                console.warn(
+                    "maxFeePerGas and maxPriorityFeePerGas are not supported on this chain as it supports Legacy Transacitons. Ignoring them."
+                );
+            }
+            return {
+                // It's on zerodev doc that we need to ignore ts errros on maxFeePerGas and maxPriorityFeePerGas
+                // https://docs.zerodev.app/sdk/faqs/use-with-gelato#transaction-configuration
+                maxFeePerGas: "0x0" as any,
+                maxPriorityFeePerGas: "0x0" as any,
+            };
+        }
+    }
```

Comment:
this file is already quite long with 400 lines–do these rely on some internal aspect or can we put these into a separate helper file?
`hasEIP1559Support(chain: Blockchain)`